lib/stronger_parameters/controller_support/permitted_parameters.rb

```diff
@@ -85,15 +85,15 @@ def permit_parameters
         end
 
         # TODO: invalid values should also be logged, but atm only invalid keys are
-        permitted_params = without_invalid_parameter_exceptions { params.permit(permitted) }
-        unpermitted_keys = flat_keys(params) - flat_keys(permitted_params)
         log_unpermitted = self.class.log_unpermitted_parameters
+        permitted_params = without_invalid_parameter_exceptions(log_unpermitted) { params.permit(permitted) }
+        unpermitted_keys = flat_keys(params) - flat_keys(permitted_params)
 
         show_unpermitted_keys(unpermitted_keys, log_unpermitted)
 
         return if log_unpermitted
 
-        params.replace(permitted_params)
+        (ActionPack::VERSION::MAJOR >= 5 ? params.send(:parameters) : params).replace(permitted_params)
```

Comment:
did you mean `params.send(:parameters, permitted_params)` ? if yes, is it feasible to get a failing test?